src/argilla/client/feedback/dataset.py

```diff
@@ -199,9 +221,10 @@ def __init__(
             raise TypeError(f"Expected `questions` to be a list, got {type(questions)} instead.")
         any_required = False
         for question in questions:
-            if not isinstance(question, (TextQuestion, RatingQuestion)):
+            if not isinstance(question, (TextQuestion, RatingQuestion, SingleLabelQuestion, MultiLabelQuestion)):
```

Comment:
could we use the `AllowedQuestion` typing somehow?

Reply:
We cannot call `isintance` over an `Union`, neither `issubclass` 😢